handler.c

```diff
@@ -1281,6 +1295,37 @@ static EFI_STATUS verify_auth_var(uint8_t *name, UINTN name_len,
     return status;
 }
 
+static bool
+check_ppi_variables(uint8_t *name, UINTN name_len, EFI_GUID *guid,uint8_t *data, UINTN data_len)
+{
+    if (name_len == sizeof(TCG2_PHYSICAL_PRESENCEFLAGSLOCK_NAME) &&
+            !memcmp(name,TCG2_PHYSICAL_PRESENCEFLAGSLOCK_NAME, name_len) &&
+            !memcmp(guid, &gEfiTcg2PpiXenGuid, GUID_LEN)) {
+       if (data_len != sizeof (uint8_t)) {
+           DBG("Bad PPI lock write. size=%ld\n", data_len);
+           return true;
+       }
+       if (*data != 1) {
+           DBG("Bad PPI lock write. data=%u\n", *data);
+           return true;
+       }
+       DBG("PPI lock set!\n");
+       ppi_locked = true;
+       return false;
+    }
+
+    if (!ppi_locked) {
+       return false;
+    }
+
+    if ((name_len == sizeof(TCG2_PHYSICAL_PRESENCEFLAGS_NAME) &&
+            !memcmp(name,TCG2_PHYSICAL_PRESENCEFLAGS_NAME,name_len)) &&
+            !memcmp(guid, &gEfiTcg2PpiXenGuid, GUID_LEN)) {
+       DBG("Attempt to set PPI flags while locked!\n");
+       return true;
```

Comment:
This function should probably return an EFI_STATUS. This should be EFI_WRITE_PROTECTED, the above bad data checks should be EFI_INVALID_PARAMETER, and the success case can be EFI_SUCCESS.

Reply:
I was just following what was agreed in the design document, based on your suggestion.
But this probebly would be better.